### ISSUE TITLE
Import a specific version of a package? [question]

### ISSUE THREAD

**Issue by cosmos72** [gopherdata/gophernotes#218]
@mattn wrote in #184 
> Sorry off topic. Can I import specific version of package in notebook?

**Comment by cosmos72** [comment]
Currently it's not possible.

My first impression is that it would require two changes:
1. a syntax to specify both the package and its version - for example something like `import "some/package@v1.2"` except that `@` should be replaced with a character that **cannot** be part of a package name
2. a mechanism to forward the package version to `go build` - putting it in the file `go.mod` of the plugin created on-the-fly and compiled to load the package could suffice

**Comment by elamre** [comment]
this would be a very appreciated addition. A space character would not work?

**Comment by cosmos72** [comment]
Directory names can contain spaces.
Actually, at least on Linux, directory names can contain **any** character except ASCII 47 (i.e. forward slash '/') and ASCII 0.

**Comment by mattn** [comment]
I prefer that gophernotes recognize go.mod file placed in same location of notebook file.

**Comment by cosmos72** [comment]
The implementation is relatively straightforward: it's just a matter of propagating the requested version through some code layers, and write it in the `go.mod` file of the plugin that will be compiled.

About the syntax: `go get` and `go install` at command line now accept the syntax `example.com/my/package@latest` or `example.com/my/package@vxx.yy.zz` so the syntax
```go
import "some/package@v1.2"
```
is now acceptable, as that's exactly what these tools expect.
Any package whose path contains '@' will now break the official tools `go get` and `go install` so there will be a strong pressure to rename them.

In summary: it's now just a matter of implementing it.
Don't expect me to have time in the next few weeks, though...
